### DISCUSSION TITLE
Deploy Dev Docker behind Traefik proxy

### DISCUSSION THREAD

**Discussion by Bbillyben** [inventree/InvenTree#3469] [Q&A] [answered]
Hi there,

I'm trying to deploy a dev env.  (branch master) behind a traefik proxy, which is already in use for other container on that VM.

The Inventree env. just work fine without traefik handling the request. 
I've done nothing except modifying the docker-compose file, to add labels for traefik :
(after compiling and adding superuser).

The DC file is (with some of my test for traefik) : 
     
```
version: "3.8"

services:
    inventree-dev-db:
        container_name: inventree-dev-db
        image: postgres:13
        expose:
            - 5432/tcp
        environment:
            - PGDATA=/var/lib/postgresql/data/pgdb
            - POSTGRES_USER=pguser
            - POSTGRES_PASSWORD=pgpassword
            - POSTGRES_DB=inventree
        volumes:
          - ./data:/var/lib/postgresql/data
        restart: unless-stopped
		labels:
          - "traefik.enable=false"

    inventree-dev-server:
        container_name: inventree-dev-server
        depends_on:
          - inventree-dev-db
        build:
            context: .
            target: dev
        image: inventree-dev-image
 #       ports:
 #           - 8000:8000
        expose:
		     -8000
        volumes:
            - ./:/home/inventree
        env_file:
            - docker.dev.env
        restart: unless-stopped
		labels:
            - "traefik.enable=true"
            - "traefik.http.routers.stock.rule=Host(`my.site.com`) && PathPrefix(`/stock`)"
            - "traefik.http.routers.stock.entrypoints= http, https"
#            - "traefik.http.routers.stock.middlewares=stock-stripprefix"
#            - "traefik.http.middlewares.stock-stripprefix.stripprefix.prefixes=/stock"
#            - "traefik.http.routers.stock.RawPath='/stock' "
#            - "traefik.http.routers.stock.entrypoints= priv"
#            - "traefik.http.services.stock.loadbalancer.server.port=8000"



    inventree-dev-worker:
        container_name: inventree-dev-worker
        image: inventree-dev-image
        command: invoke worker
        depends_on:
            - inventree-dev-server
        volumes:
            - ./:/home/inventree
        env_file:
            - docker.dev.env
        restart: unless-stopped
		labels:
          - "traefik.enable=false"
```

But I always land on a 504 Gateway Timeout : 
`time="2022-08-04T09:57:59Z" level=debug msg="'504 Gateway Timeout' caused by: dial tcp 192.168.64.3:8000: i/o timeout"`

Is there something I missed ? or any clue? 

___________________________________________________ 

Docker version 20.10.17, build 100c701
Docker Compose version v2.0.1  (I need that version for other project, but it has worked with latest branch of Inventree)
Debian GNU/Linux 11 (bullseye) / Linux 5.10.0-15-amd64

**Comment by miggland** [comment, marked as answer]
Do you have the Traefik proxy on the same docker network as InvenTree?

I suggest to try without "/stock" completely, to make sure you get it to work without making any adjustments to paths.

**Reply by matmair** [reply]
We do not support hosting on subdirs. Please use only subdomains. Ref #1297

**Reply by matmair** [reply]
Hi there @Bbillyben @miggland I would appreciate it if you would do your discussions as replies and not seperate answers as that makes it hard to follow the dicussion and will break once answers get reordered.

**Comment by Bbillyben** [comment]
Shame on me..... thanks @miggland , that was a misspelled network name.... 

I'm now stucked on a 302 reponse... but it's moving on!

**Comment by miggland** [comment]
Good, then keep trying! :+1:

Do you actually want to be able to use both http and https entrypoints? Or do you want redirection to https if the http address is called?

If the second is the case, you should put the redirection in the entrypoint definition, ie on the Traefik container, and not on InvenTree.

Something like this should be in the Traefik config:
 ```
      - --entrypoints.web.address=:80
      - --entrypoints.web.http.redirections.entryPoint.to=websecure
      - --entrypoints.web.http.redirections.entryPoint.scheme=https
      - --entrypoints.websecure.address=:443
```
and your InvenTree entrypoint is then:
```
- "traefik.http.routers.inventoryweb.entrypoints=websecure"
```

**Comment by Bbillyben** [comment]
Thanks migglant, everything is redirected to 443 in the environment of the vm!

I've [edit do not] solved my issue, but really not ellegantly!

Is there anywhere to set the 'Base URL' parameter except from inventree app directly? 

=> I've done it by making the docker redirect to the root url to access inventree, set the base URL in the setting and revert it to the '/stock' sub url.

it works, but it's not something I would do for the 'prod' step!


[edit] Ok i have to work a lilttle more, there is no redirection using base url...

**Reply by matmair** [reply]
@Bbillyben we do not support subdirs so this will not work. Ref #1297

**Reply by Bbillyben** [reply]
thanks, I will follow that issue!

**Reply by matmair** [reply]
@Bbillyben would you be up to submit a sample Traefik config to the docs? I think that could be helpful for the community but I have never touched Traefik.

**Reply by Bbillyben** [reply]
I will try to set a std config up to share with some relevant comment.
